src/coreclr/vm/frames.cpp

```diff
@@ -37,6 +37,261 @@
 
 #define CHECK_APP_DOMAIN    0
 
+#ifdef DACCESS_COMPILE
+#define FRAME_POLYMORPHIC_DISPATCH_UNREACHABLE() _ASSERTE("!Unexpected value in Frame")
+#else
+#define FRAME_POLYMORPHIC_DISPATCH_UNREACHABLE() DoJITFailFast()
+#endif
+
+void Frame::GcScanRoots(promote_func *fn, ScanContext* sc)
+{
+    switch (this->GetFrameIdentifier())
```

Comment:
```suggestion
    switch (GetFrameIdentifier())
```
Nit: `this->` should not be needed